spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories

```diff
@@ -15,7 +15,8 @@ org.springframework.cloud.bootstrap.BootstrapConfiguration=\
 org.springframework.cloud.netflix.config.DiscoveryClientConfigServiceBootstrapConfiguration
 
 org.springframework.cloud.client.discovery.EnableDiscoveryClient=\
-org.springframework.cloud.netflix.eureka.EurekaDiscoveryClientConfiguration
+org.springframework.cloud.netflix.eureka.EurekaDiscoveryClientConfiguration,\
+org.springframework.cloud.netflix.eureka.EurekaApplicationEventAutoConfiguration
```

Comment:
I was just going to submit a separate PR to do the opposite.  `EurekaClientAutoConfiguration` should probably move down to run only under `EnableDiscoveryClient`.  As it stands right now, `EurekaClient` is initializing even when `@EnableEureka` is not applied, and requires an extra `eureka.client.enabled=false` in properties.  This is my fault - stemming from the work on bootstrapping `EurekaClient` before SmartLifecycle phase 0.